File: src/conf/plugins/coding/luasnip-snippets.ts
import { Plugin, PluginOptsBase } from "@core/model";

const spec: PluginOptsBase = {
  shortUrl: "TwIStOy/luasnip-snippets",
  lazy: {
    dev: true,
    dependencies: ["L3MON4D3/LuaSnip"],
    event: ["InsertEnter"],
    config: () => {
      luaRequire("luasnip-snippets").setup({
        user: {
          name: "Hawtian Wang",
        },
      });
    },
  },
};

export const plugin = new Plugin(spec);

File: lua/dotvim/plugins/coding/luasnip-snippets.lua
---@type LazyPluginSpec
return {
  "TwIStOy/luasnip-snippets",
  event = { "InsertEnter" },
  dependencies = "nvim-treesitter/nvim-treesitter",
  opts = {
    user = {
      name = "Hawtian Wang",
    },
    snippet = {
      lua = {
        vim_snippet = true,
      },
      rust = {
        rstest_support = true,
      },
      cpp = {
        quick_type = {
          extra_trig = {
            { trig = "m", params = 2, template = "std::unordered_map<%s, %s>" },
          },
          qt = true,
          cpplint = false,
        },
      },
    },
  },
}
